pkg/apis/eventing/v1alpha1/provisioner_types.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"encoding/json"
+	"github.com/knative/pkg/apis"
+	"github.com/knative/pkg/webhook"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Provisioner provisions.
+type Provisioner struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ProvisionerSpec   `json:"spec"`
+	Status ProvisionerStatus `json:"status"`
+}
+
+// Check that Provisioner can be validated and can be defaulted.
+var _ apis.Validatable = (*Provisioner)(nil)
+var _ apis.Defaultable = (*Provisioner)(nil)
+var _ runtime.Object = (*Provisioner)(nil)
+var _ webhook.GenericCRD = (*Provisioner)(nil)
+
+// ProvisionerSpec is the spec for a Provisioner resource.
+type ProvisionerSpec struct {
+	// TODO: Generation does not work correctly with CRD. They are scrubbed
```

Comment:
Is this still the case? I recall that issue was closed awhile ago, but each new CRD we create carries this cut&paste around, could we validate this is still necessary?

Reply:
It is still valid until we change some code that needs generation on spec. @mattmoor told me it was still an issue. maybe related until we stop supporting older k8s versions?

Reply:
Ok, well, let's figure out what the correct TODO here is and update the comment that now seems to be inaccurate.

Reply:
I talked with @mattmoor -- the short version is that we'll need this for the next ~2 months:

Kubernetes 1.11 (rolling out in commercial providers SOON™) enables the `/status` subresource, which will allow us to start rewriting our controllers to perform updates via `/status`. Once all our controllers are using `/status`, we can ignore the `spec.Generation` field. It's not clear we'll be able to remove it from the schema until the next API version after CRD versioning is deployed.